src/sdk/service/network-validation-service/network-validation-service.ts

```diff
@@ -0,0 +1,333 @@
+import { Netmask, IPv4AddressRange, IPv4NetworkDescriptor, IPv4Address, Cidr } from './network-validation-types';
+
+/**
+ * Network Validation Service
+ */
+export abstract class NetworkValidationService {
+  /**
+   * Gets the integer representation of an IP address.
+   * @param {string} ip the ip address.
+   * @return {number} The integer representation.
+   */
+  public static getIPv4AsInteger(ipAddress: IPv4Address | Cidr): number | null {
+    const components = ipAddress.address.split('.');
+    let combined = 0;
+    for (let i = components.length - 1; i >= 0; i--) {
+      const index = 3 - i;
+      const component = Number(components[index]);
+      combined = combined | (component << (i * 8));
+    }
+    return combined;
+  }
+
+  /**
+   * Gets prefix size of net mask.
+   * @param {string} netmask netmask string.
+   * @return {number} prefix size of netmask.
+   */
+  public static getNetmaskPrefixSize(netmask: IPv4Address): number | null {
+    const maskInt = this.getIPv4AsInteger(netmask);
+    const template = parseInt('80000000', 16);
+    for (let i = 0; i <= 32; i++) {
+      if ((template >> (i - 1)) === maskInt) {
+        return i;
+      }
+    }
+  }
+
+  /**
+   * Gets the netmask and gateway/ip from a CIDR formatted network string.
+   * @param {string} cidr cidr string.
+   * @return {IPv4NetworkDescriptor} of the form { gateway: '10.10.10.0', netmask:
+   *         '255.255.255.0' }
+   */
+  public static getIPv4NetworkDescriptorFromCidr(cidr: Cidr): IPv4NetworkDescriptor | null {
+    const mask = parseInt('80000000', 16) >> (cidr.hostIdentifier - 1);
+    const netmask = new Netmask(this.getIntegerAsIPv4Address(mask));
+    return {
+      gateway: cidr.address,
+      netmask: netmask.address
+    };
+  }
+
+  /**
+   * Gets an IP range object from a string of the format: a.b.c.d - e.f.g.h
+   * @param {string} rangeStr: the range string to parse
+   * @return {IPv4AddressRange} range object of the form stated in the description or null if invalid.
+   */
+  public static getIPv4AddressRangeFromString(rangeStr: string): IPv4AddressRange | null {
```

Comment:
would be nice to have a separate, public method for checking whether a range string is valid and then use that in here.